File: go/countdown.go
package main

import (
	"fmt"
	"time"
)

const Number uint = 1e9 // 1 Billion

func countdown(n uint) {
	for n > 0 {
		n = n - 1
	}
	return

}

func main() {

	fmt.Printf("Counting down %d\n", Number)

	start := time.Now()
	countdown(Number)
	end := time.Now()

	fmt.Printf("Took %v\n", end.Sub(start))

}

File: rust/countdown.rs
use std::time::{Instant};

static NUMBER: u64 = 1_000_000_000_000_000_000;

fn countdown(num: u64) {
    let mut i = num;
    while i > 0 {
        i -= 1;
    }
}

fn main() {
    println!("Counting down {:?}", NUMBER);
    let start_time = Instant::now();
    countdown(NUMBER);
    let elapsed_time = start_time.elapsed();
    println!("Took {:?}", elapsed_time);
}

